src/core/reaction_ensemble.cpp

```diff
@@ -1,1724 +1,1970 @@
+                                   // CONSTRAINTS here, but only interesting,
+                                   // when doing MD/ HMC because then the system
+                                   // might explode easily here due to high
+                                   // forces
+      insert_tries += 1;
+      if (d_min > exclusion_radius)
+        particle_inserted_too_close_to_another_one = false;
     }
-	double pos_vec[3];
-	
-	//create random velocity vector according to Maxwell Boltzmann distribution for components
-	double vel[3];
-	//we usse mass=1 for all particles, think about adapting this
-	vel[0]=std::pow(2*PI*m_current_reaction_system.temperature_reaction_ensemble,-3.0/2.0)*gaussian_random()*time_step;//scale for internal use in espresso
-	vel[1]=std::pow(2*PI*m_current_reaction_system.temperature_reaction_ensemble,-3.0/2.0)*gaussian_random()*time_step;//scale for internal use in espresso
-	vel[2]=std::pow(2*PI*m_current_reaction_system.temperature_reaction_ensemble,-3.0/2.0)*gaussian_random()*time_step;//scale for internal use in espresso
-	double charge= (double) m_current_reaction_system.charges_of_types[find_index_of_type(desired_type)];
-	bool particle_inserted_too_close_to_another_one=true;
-	int max_insert_tries=1000;
-	int insert_tries=0;
-	double min_dist=m_current_reaction_system.exclusion_radius; //setting of a minimal distance is allowed to avoid overlapping configurations if there is a repulsive potential. States with very high energies have a probability of almost zero and therefore do not contribute to ensemble averages.
-	if(min_dist!=0){
-		while(particle_inserted_too_close_to_another_one && insert_tries<max_insert_tries) {
-			get_random_position_in_box(pos_vec);
-			place_particle(p_id,pos_vec);
-			//set type
-			set_particle_type(p_id, desired_type);
-			#ifdef ELECTROSTATICS
-			//set charge
-			set_particle_q(p_id, charge);
-			#endif
-			//set velocities
-			set_particle_v(p_id,vel);
-			double d_min=distto(partCfg(), pos_vec,p_id); //TODO also catch constraints with an IFDEF CONSTRAINTS here, but only interesting, when doing MD/ HMC because then the system might explode easily here due to high forces
-			insert_tries+=1;
-			if(d_min>m_current_reaction_system.exclusion_radius)
-				particle_inserted_too_close_to_another_one=false;
-		}
-	}else{
-		get_random_position_in_box(pos_vec);
-		place_particle(p_id,pos_vec);
-		//set type
-		set_particle_type(p_id, desired_type);	
-		//set velocities
-		set_particle_v(p_id,vel);
-		#ifdef ELECTROSTATICS
-		//set charge
-		set_particle_q(p_id, charge);
-		#endif
-	}
-	if(insert_tries>max_insert_tries){
-		throw std::runtime_error("No particle inserted");
-		return -1;	
-	}
-	return p_id;
+  } else {
+    get_random_position_in_box(pos_vec);
+    place_particle(p_id, pos_vec);
+    // set type
+    set_particle_type(p_id, desired_type);
+    // set velocities
+    set_particle_v(p_id, vel);
+#ifdef ELECTROSTATICS
+    //
```

Comment:
the next three functions are unrelated to the reaction_ensemble and should be moved to an appropriate utils header